docs/src/proposals/gasless-evm.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Gasless EVM
+---
+
+## Problem
+
+Applications targeting other blockchains use audited smart contracts
+implemented in Solidity. Running that same application targeting Solana
+requires migrating the smart contract to Rust and the Sealevel parallel
+programming model. Creating the same smart contract in Solana requires
+mechanical transformations of high-level Solidity constructs to lower-level
+Rust equivalents. The process is tedious and allows for human error, requiring
+a second audit of the new code. Automating the process would save time and
+alleviate the need for an audit.
+
+## Proposed Solution
+
+Add an EVM bytecode loader. Map Solidity contract storage to Solana account
+data. Map runtime data to Solana sysvars. Require transactions to specify which
+accounts they may access, such that Sealevel may execute independent contracts
+in parallel. Instead of requiring "gas", map EVM bytecode to BPF so that
+contracts are subject to the BPF instruction limit. Support Solidity method
+calls from arbitrary Solana programs and executing Solana instructions from
+Solidity methods.
+
+### Developer Experience
+
+```bash
+solc --bin sourceFile.sol > lib-hello.bin
+solana-keygen new -o lib-hello.json
+solana deploy lib-hello.json lib-hello.bin
+```
+
+```bash
+solana-keygen new -o hello-world.json
+solana-evm lib-hello.json HelloWorld "Hello world!" -a hello-world.json
+solana-evm lib-hello.json update "Goodbye cruel world!" -a hello-world.json
+```
+
+### EvmInstruction
+
+```
+///   0. `[]` Clock sysvar
```

Comment:
Might be able to push this into the contract accounts. If you don't use TIMESTAMP, then you shouldn't have to burn 32 bytes of transaction space passing this in.